### PR TITLE
teams: don't store team if no change

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

**Comment by maxtaco** [comment]
cc @mmaxim

File: go/teams/loader.go [hunk 1]
```diff
@@ -905,9 +905,12 @@ func (l *TeamLoader) load2InnerLockedRetry(ctx context.Context, arg load2ArgT) (
 		return nil, err
 	}
 
-	// Cache the validated result
-	tracer.Stage("put")
-	l.storage.Put(mctx, ret)
+	// Cache the validated result if it was actually updated via the team/get endpoint. In many cases, we're not
+	// actually mutating the team.
+	if teamUpdate != nil {
```
Comment (mlsteele) [inline go/teams/loader.go]: maybe `teamUpdate != nil || didRepoll`

Comment (mlsteele) [reply go/teams/loader.go]: Also `fillInStubbedLinks` should get saved. Not sure this covers that yet

File: go/teams/storage/merkle.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+// Increment to invalidate the disk cache.
+const merkleDiskStorageVersion = 1
+const merkleMemCacheLRUSize = 2000
+
+type merkleDiskStorageItem struct {
+	Version  int           `codec:"V"`
+	PolledAt keybase1.Time `codec:"T"`
+}
+
+func NewMerkle() *Merkle {
```
Comment (mlsteele) [inline go/teams/storage/merkle.go]: Anything special that makes this not able to use storage/generic?

**Pull request merged**